pkg/sdk/testint/tasks_gen_integration_test.go

```diff
@@ -12,324 +14,328 @@ import (
-		assert.Empty(t, task.Condition)
-		assert.Empty(t, task.AllowOverlappingExecution)
-		assert.Empty(t, task.ErrorIntegration)
-		assert.Empty(t, task.LastCommittedOn)
-		assert.Empty(t, task.LastSuspendedOn)
-		assert.Empty(t, task.OwnerRoleType)
-		assert.Empty(t, task.Config)
-		assert.Empty(t, task.Budget)
+		assertions.AssertThat(t, objectassert.TaskFromObject(t, task).
+			HasNotEmptyCreatedOn().
+			HasName(id.Name()).
+			HasDatabaseName(testClientHelper().Ids.DatabaseId().Name()).
+			HasSchemaName(testClientHelper().Ids.SchemaId().Name()).
+			HasSchedule(schedule).
+			// all below are not contained in the terse response, that's why all of them we expect to be empty
+			HasId("").
+			HasOwner("").
+			HasComment("").
+			HasWarehouse("").
+			HasPredecessors().
+			HasState("").
+			HasDefinition("").
+			HasCondition("").
+			HasAllowOverlappingExecution(false).
+			HasErrorIntegration("").
+			HasLastCommittedOn("").
+			HasLastSuspendedOn("").
+			HasOwnerRoleType("").
+			HasConfig("").
+			HasBudget("").
+			HasLastSuspendedOn("").
+			HasTaskRelations(sdk.TaskRelations{}),
+		)
 	}
 
-	cleanupTaskProvider := func(id sdk.SchemaObjectIdentifier) func() {
-		return func() {
-			err := client.Tasks.Drop(ctx, sdk.NewDropTaskRequest(id))
-			require.NoError(t, err)
-		}
-	}
-
-	createTaskBasicRequest := func(t *testing.T) *sdk.CreateTaskRequest {
-		t.Helper()
+	t.Run("create task: no optionals", func(t *testing.T) {
 		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
 
-		return sdk.NewCreateTaskRequest(id, sql)
-	}
-
-	createTaskWithRequest := func(t *testing.T, request *sdk.CreateTaskRequest) *sdk.Task {
-		t.Helper()
-		id := request.GetName()
-
-		err := client.Tasks.Create(ctx, request)
+		err := testClient(t).Tasks.Create(ctx, sdk.NewCreateTaskRequest(id, sql))
 		require.NoError(t, err)
-		t.Cleanup(cleanupTaskProvider(id))
+		t.Cleanup(testClientHelper().Task.DropTaskFunc(t, id))
 
-		task, err := client.Tasks.ShowByID(ctx, id)
+		task, err := testClientHelper().Task.Show(t, id)
 		require.NoError(t, err)
 
-		return task
-	}
-
-	createTask := func(t *testing.T) *sdk.Task {
-		t.Helper()
-		return createTaskWithRequest(t, createTaskBasicRequest(t))
-	}
-
-	t.Run("create task: no optionals", func(t *testing.T) {
-		request := createTaskBasicRequest(t)
-
-		task := createTaskWithRequest(t, request)
-
-		assertTask(t, task, request.GetName())
+		assertTask(t, task, id, "")
 	})
 
 	t.Run("create task: with initial warehouse", func(t *testing.T) {
-		request := createTaskBasicRequest(t).
-			WithWarehouse(sdk.NewCreateTaskWarehouseRequest().WithUserTaskManagedInitialWarehouseSize(sdk.Pointer(sdk.WarehouseSizeXSmall)))
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
 
-		task := createTaskWithRequest(t, request)
+		err := testClient(t).Tasks.Create(ctx, sdk.NewCreateTaskRequest(id, sql).WithWarehouse(*sdk.NewCreateTaskWarehouseRequest().WithUserTaskManagedInitialWarehouseSize(sdk.WarehouseSizeXSmall)))
```

Comment:
This warehouse size param is not checked.

Reply:
Because it can't be checked. I left this test because it showed that we can create tasks with this value set, but the value, sadly, cannot be checked.

Reply:
Update, it's in session parameters, added a check